tests/test-grad0.c

```diff
@@ -372,11 +372,10 @@ bool check_mat_mul(
 #define NUM_PERMUTATIONS (4*3*2*1)
 
 int main(int argc, const char ** argv) {
-    struct ggml_init_params params = {
-        .mem_size   = 128*1024*1024,
-        .mem_buffer = NULL,
-        .no_alloc   = false,
-    };
+    struct ggml_init_params params;
+    params.mem_size   = 128*1024*1024;
+    params.mem_buffer = NULL;
+    params.no_alloc   = false;
```

Comment:
Why though? Pretty sure this is standard in C99 and this is a C file, not cpp.

Reply:
test-double-float, test-grad0, and test-opt are all .c files that are being built with g++, and linked to llama.o and common.o which are C++ objects. Either their build and link steps need to be separated so they can be built with gcc, or they should be ported to C++.

Reply:
Use initializer list with member names in comments - see other use cases in llama.cpp

Reply:
I think we should at least rename the files to end in .cpp to reflect the language they are actually being compiled as.

Reply:
That’s fine but we still want to use initialiser list as so we get warnings upon struct changes